src/app/services/token-lifetime.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Inject, Injectable } from '@angular/core';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { Store } from '@ngrx/store';
+import { TranslateService } from '@ngx-translate/core';
+import { format } from 'date-fns';
+import { filter } from 'rxjs';
+import { WINDOW } from 'app/helpers/window.helper';
+import { Timeout } from 'app/interfaces/timeout.interface';
+import { JobSlice, selectJobsPanelSlice } from 'app/modules/jobs/store/job.selectors';
+import { WebSocketService } from 'app/services';
+import { DialogService } from 'app/services/dialog.service';
+import { AppState } from 'app/store';
+import { selectPreferences } from 'app/store/preferences/preferences.selectors';
+
+@UntilDestroy()
+@Injectable({
+  providedIn: 'root',
+})
+export class TokenLifetimeService {
+  protected actionWaitTimeout: Timeout;
+  protected terminateCancelTimeout: Timeout;
+  private startBound;
+  private jobsCount = 0;
+
+  constructor(
+    private dialogService: DialogService,
+    private translate: TranslateService,
+    private ws: WebSocketService,
+    private appStore$: Store<AppState>,
+    private jobStore$: Store<JobSlice>,
+    @Inject(WINDOW) private window: Window,
+  ) {
+    this.startBound = this.start.bind(this);
+
+    this.jobStore$.select(selectJobsPanelSlice).pipe(untilDestroyed(this)).subscribe((jobs) => {
+      if (jobs.length) {
+        this.stop(true);
+      } else if (this.jobsCount) {
+        this.start(true);
+      }
+      this.jobsCount = jobs.length;
+    });
+  }
+
+  start(addListeners = false): void {
```

Comment:
1. This method is a bit hard to read, can we break it apart?
2. Can we refactor to not have an argument? I.e. check if listeners are already set, or add new separate methods called `pause` and `resume`?